core/src/main/java/org/infinispan/marshall/core/internal/StreamBytesObjectOutput.java

```diff
@@ -0,0 +1,103 @@
+package org.infinispan.marshall.core.internal;
+
+import java.io.IOException;
+import java.io.ObjectOutput;
+import java.io.OutputStream;
+
+final class StreamBytesObjectOutput implements ObjectOutput {
```

Comment:
How common is it to have a `BytesObjectOutput` that's not wrapped in a `StreamBytesObjectOutput`? Memory-wise, I think it would be cheaper to have a `stream` field in every `BytesObjectOutput`.